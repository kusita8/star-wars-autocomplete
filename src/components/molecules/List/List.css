.list {
  border: 1px solid black;
}

.list__empty-results {
  height: 3rem;
  padding: 0 0.4rem;
  display: flex;
  align-items: center;
}

.list--loading {
  height: 20rem;
  background: linear-gradient(90deg, hsl(210, 15%, 88%), hsl(210, 15%, 95%), hsl(210, 15%, 88%));
  background-size: 200%;
  animation: skeletonAnimation 1s infinite;
  position: relative;
}

@keyframes skeletonAnimation {
  from {
    background-position: -100% 0;
  }
  to {
    background-position: 100% 0;
  }
}
