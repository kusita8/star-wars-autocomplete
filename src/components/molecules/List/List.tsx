import { FC, useMemo } from "react";
import { Person } from "../../../types/Person.interface";
import { getClassNames } from "../../../utils/getClassNames";
import { ListItem } from "../../atoms/ListItem/ListItem";
import "./List.css";

interface ListProps {
  persons?: Person[];
  loading?: boolean;
  search?: string;
}

export const List: FC<ListProps> = ({ persons, loading, search }) => {
  const content = useMemo(() => {
    if (loading) return null;

    if (persons && persons.length > 0) {
      return persons?.map((person) => (
        <ListItem search={search} key={person.name} person={person} />
      ));
    }

    return <p className="list__empty-results">No results found for {search}</p>;
  }, [loading, persons]);

  return <div className={getClassNames("list", loading && "list--loading")}>{content}</div>;
};
