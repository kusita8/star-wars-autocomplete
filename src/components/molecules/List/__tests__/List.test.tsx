import { render } from "@testing-library/react";
import { List } from '../List';

describe('List', () => {  
  test('it should match the snapshot', () => {
    const { asFragment } = render(<List />);
    expect(asFragment()).toMatchSnapshot();
  });
});