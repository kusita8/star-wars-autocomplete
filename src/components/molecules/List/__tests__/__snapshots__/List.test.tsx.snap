// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <p
      class="list__empty-results"
    >
      No results found for 
    </p>
  </div>
</DocumentFragment>
`;
